libs/h3/src/main/java/org/elasticsearch/h3/Vec2d.java

```diff
@@ -320,99 +312,4 @@ static double posAngleRads(double rads) {
             return rads;
         }
     }
-
-    /**
-     * Computes the point on the sphere a specified azimuth and distance from
-     * another point.
-     *
-     * @param p1       The first spherical coordinates.
-     * @param az       The desired azimuth from p1.
-     * @param distance The desired distance from p1, must be non-negative.
-     *                 p1.
-     */
-    private static LatLng geoAzDistanceRads(LatLng p1, double az, double distance) {
-        if (distance < Constants.EPSILON) {
-            return p1;
-        }
-
-        double sinlat, sinlng, coslng;
-
-        az = posAngleRads(az);
-
-        double lat, lon;
-
-        // check for due north/south azimuth
-        if (az < Constants.EPSILON || Math.abs(az - M_PI) < Constants.EPSILON) {
-            if (az < Constants.EPSILON) {// due north
```

Comment:
I removed all this optimisations in LatLng because there seems to never be called during the tests and they actually produce bad results if `p1.getLatRad() + distance > Math.PI /2 ` or `p1.getLatRad() - distance < -Math.PI/2`